iml-agent/src/action_plugins/lpurge.rs

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2019 DDN. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+use crate::agent_error::ImlAgentError;
+
+use std::fmt;
+use std::path::Path;
+
+#[derive(serde::Deserialize)]
+pub struct Config {
+    fs_name: String,
+    ost_name: String,
+    pool_name: String,
+    freelo: u8,
+    freehi: u8,
+}
+
+impl fmt::Display for Config {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(
+            f,
+            "\
+             device={fs_name}-{ost_name}\n\
+             dryrun=true\n\
+             freehi={freehi}\n\
+             freelo={freelo}\n\
+             listen_socket=/run/lpurge-{fs_name}-{ost_name}-{pool_name}.sock\n\
+             max_jobs=0\n\
+             pool={fs_name}.{pool_name}\n\
+             ",
+            freehi = self.freehi,
+            freelo = self.freelo,
+            fs_name = self.fs_name,
+            ost_name = self.ost_name,
+            pool_name = self.pool_name,
+        )
+    }
+}
+
+pub async fn create_lpurge_conf(c: Config) -> Result<(), ImlAgentError> {
+    write("/etc/lpurge", &c).map_err(|e| e.into())
+}
+
+fn write<P: AsRef<Path>>(dir: P, c: &Config) -> std::io::Result<()> {
+    std::fs::create_dir_all(&dir)?;
```

Comment:
This needs to use the async variant from tokio.

Reply:
Yes, underway.